viewing_party/party.py

```diff
@@ -1,23 +1,360 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    '''
+    create_movie(title, genre, rating):
+    
+    takes three parameters: `title`, `genre`, `rating`
+    If those three attributes are truthy, then return a dictionary. This dictionary should...
+    Have three key-value pairs, with specific keys
+    The three keys should be `"title"`, `"genre"`, and `"rating"`
+    The values of these key-value pairs should be appropriate values
+    If `title` is falsy, `genre` is falsy, or `rating` is falsy, this function should return `None`
+
+    returns movie_dictionary
+    '''
+    if (not title) or (not genre) or (not rating):
+        # here we check if we have any of the attributes are falsy, we return None 
+        return None
+
+    movie_dictionary = {} #create empty dictionary
+
+    movie_dictionary["title"] = title #add key "title" with corresponding value
+    movie_dictionary["genre"] = genre #add key "genre" with corresponding value
+    movie_dictionary["rating"] = rating #add key "rating" with corresponding value
+   
+    return movie_dictionary
+
+def add_to_watched(user_data, movie):
+    '''
+    - take two parameters: `user_data`, `movie`        
+    - add the `movie` to the `"watched"` list inside of `user_data`
+    - return the `user_data`    
+    '''
+
+    #appending the dictionary "movie" to the list user_data["watched"]
+    user_data["watched"].append(movie) 
+    return user_data
+    
+def add_to_watchlist(user_data, movie):       
+    '''
+    - take two parameters: `user_data`, `movie`        
+    - add the `movie` to the `"watchlist"` list inside of `user_data`
+    - return the `user_data`    
+    '''    
+
+    #appending the dictionary "movie" to the list user_data["watchlist"]
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    - take two parameters: `user_data`, `title`
+    - If the title is in a movie in the user's watchlist:
+        - remove that movie from the watchlist
+        - add that movie to watched
+        - return the `user_data`
+    - If the title is not a movie in the user's watchlist:
+        - return the `user_data`
+    '''
+    # using the for loop to iterate through the list user_data["watchlist"]
+    for movie in user_data["watchlist"]:
+        # and check if the movie's title is what we're looking for,    
+        if movie["title"] == title:
+            # then we add it to the "watched"            
+            user_data["watched"].append(movie)
+            # and remove it from the "watchlist"           
+            user_data["watchlist"].remove(movie)           

```

Comment:
we could add a second identical return statement to the end of this if block. That will pop us out of the function as soon as we have found the movie we are looking for. That will keep us from having to go through the rest of the list once we have found it. Note, we will still need the outer return statement in case the title doesn't exist.